tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,90 @@
+        const month = date.toLocaleString('en-US', { month: '2-digit' })
+        const day = date.toLocaleString('en-US', { day: '2-digit' })
+        const expectedDermatologyAppointmentDate = `${year}-${month}-${day}`
+        await expect(page.locator('.mat-datepicker-input')).toHaveValue(`${year}/${month}/${day}`)
+        await page.locator('#description').fill('dermatology visit')
+        await page.getByRole('button', { name: 'Add Visit' }).click()
+        await expect(petsAndVisitsRowSamatha.locator('app-visit-list tr td').first()).toHaveText(expectedDermatologyAppointmentDate)
+        await petsAndVisitsRowSamatha.getByText('Add Visit').click()
+        await page.getByLabel('Open calendar').click()
+        date.setDate(date.getDate() - 45)
+        const previousAppointmentYear = date.getFullYear();
+        const previousAppointmentMonth = date.toLocaleString('en-US', { month: '2-digit' })
+        const previousAppointmentDayNumeric = date.toLocaleString('en-US', { day: 'numeric' })
+        const previousAppointmentDay2Digit = date.toLocaleString('en-US', { day: '2-digit' })
+        const expectedMassageAppointmentDate = `${previousAppointmentYear}-${previousAppointmentMonth}-${previousAppointmentDay2Digit}`;
+        let calendarMonthAndYear = await page.locator('.mat-calendar-period-button').textContent()
+        while (!calendarMonthAndYear?.includes(`${previousAppointmentMonth} ${previousAppointmentYear}`)) {
+            await page.locator('.mat-calendar-previous-button').click()
+            calendarMonthAndYear = await page.locator('.mat-calendar-period-button').textContent()
+        }
+        await page.getByText(previousAppointmentDayNumeric, { exact: true }).click()
+        await page.locator('#description').fill('massage therapy')
+        await page.getByText('Add Visit').click()
+        const petAllAppointments = petsAndVisitsRowSamatha.locator('app-visit-list tr')
+        const laterAppointmentDateforSamathasPet = await petAllAppointments.nth(1).locator('td').first().textContent()
+        const currentAppointmentDateforSamathasPet = await petAllAppointments.nth(2).locator('td').first().textContent()
+        expect(Date.parse(currentAppointmentDateforSamathasPet!) < Date.parse(laterAppointmentDateforSamathasPet!)).toBeTruthy()
+        await petsAndVisitsRowSamatha.locator('app-visit-list tr', { hasText: expectedDermatologyAppointmentDate }).getByText('Delete Visit').click()
+        await petsAndVisitsRowSamatha.locator('app-visit-list tr', { hasText: expectedMassageAppointmentDate }).getByText('Delete Visit').click()
+        await page.waitForResponse("https://petclinic-api.bondaracademy.com/petclinic/api/visits/*")
+        for (const row of await petAllAppointments.all()) {
+            expect(await row.textContent()).not.toContain(expectedDermatologyAppointmentDate)
+            expect(await row.textContent()).not.toContain(expectedMassageAppointmentDate)
+        }
```

Comment:
Here alternatively you could use this approach. Instead of looping through the rows getting the text content for each row individually, you can get all text content for all rows, which would return array of values. Then validate, if this array have do don't have the value.
```
const samanthaVisitDates = await petAllAppointments.allTextContents()
expect(samanthaVisitDates).not.toContain(expectedDermatologyAppointmentDate)
expect(samanthaVisitDates).not.toContain(expectedMassageAppointmentDate)
```